src/openrct2/interface/Viewport.cpp

```diff
@@ -620,7 +620,7 @@ void viewport_update_position(rct_window* window)
         y += viewport->viewPos.y;
     }
 
-    viewport_move(x, y, window, viewport);
+    viewport_move({ x, y }, window, viewport);
```

Comment:
Can you please convert the `x` and `y` on this function to a single `ScreenCoordsXY` variable and use it here?

Reply:
And preferrably make it two variables one for the scope L585 to L623 and another for L546 to L549 (viewportMidPoint)